crates/orchestrator/src/data_storage/aws_s3/mod.rs

```diff
@@ -0,0 +1,130 @@
+use crate::data_storage::aws_s3::config::AWSS3Config;
+use crate::data_storage::types::StarknetOsOutput;
+use crate::data_storage::DataStorage;
+use async_trait::async_trait;
+use aws_sdk_s3::config::{Builder, Credentials, Region};
+use aws_sdk_s3::primitives::ByteStream;
+use aws_sdk_s3::{Client, Error};
+
+pub mod config;
+
+const SNOS_OUTPUT_FILE_NAME: &str = "snos_output.json";
+const KZG_FILE_NAME: &str = "kzg.txt";
+
+pub struct AWSS3 {
+    client: Client,
+    config: AWSS3Config,
+}
+
+impl AWSS3 {
+    #[allow(dead_code)]
+    pub async fn new(config: AWSS3Config) -> Self {
+        // AWS cred building
+        let credentials = Credentials::new(
+            config.s3_key_id.clone(),
+            config.s3_key_secret.clone(),
+            None,
+            None,
+            "loaded_from_custom_env",
+        );
+        let region = Region::new(config.s3_bucket_region.clone().to_string());
+        let conf_builder = Builder::new()
+            .region(region)
+            .credentials_provider(credentials);
+        let conf = conf_builder.build();
+
+        // Building AWS S3 config
+        let client = Client::from_conf(conf);
+
+        Self { client, config }
+    }
+}
+
+#[async_trait]
+impl DataStorage for AWSS3 {
+    async fn get_snos_data_for_block(&self, block_number: u128) -> Result<StarknetOsOutput, Error> {
```

Comment:
same as above, S3 should have no idea on what snos is. s3 should only have get and put functions.